tla-types/src/main/scala/at/forsyte/apalache/tla/typecmp/raw/RawLeafBuilder.scala

```diff
@@ -0,0 +1,37 @@
+package at.forsyte.apalache.tla.typecmp.raw
+
+import at.forsyte.apalache.tla.lir._
+import at.forsyte.apalache.tla.lir.values._
+
+/**
+ * Raw builder for names and literals (IR tree leaves)
+ *
+ * @author
+ *   Jure Kukovec
+ */
+trait RawLeafBuilder {
```

Comment:
```suggestion
trait RawLiteralAndNameBuilder {
```

I would prefer a descriptive name, especially if it's used only internally.